Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM registry.access.redhat.com/rhscl/ruby-23-rhel7
+
+# sshd
+USER root
+RUN yum install -y --setopt=tsflags=nodocs openssh-server libicu-devel && \
+    yum clean all && /usr/sbin/sshd-keygen
+
+RUN mkdir /var/run/sshd
+
+# TODO: is this needed? https://git.io/vrplM
+RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+# gems
+RUN ["bash", "-c", "gem install --no-ri --no-rdoc bunny"]
+
+# git user
+RUN groupadd -r git &&\
+     useradd -r -s /bin/bash -g git git && \
+     mkdir --parent /home/git && \
+     chown -R git:git /home/git
+
+# gitlab-shell setup
+COPY . /home/git/gitlab-shell
+WORKDIR /home/git/gitlab-shell
+RUN ["bash", "-c", "./bin/install"]
+
+COPY authorized_keys /home/git/.ssh/authorized_keys
+
+USER root
```

Comment:
@rhcarvalho I haven't had a chance to try it on OpenShift. I'm vaguely aware of the changes that need to be made for that, since we had to do them before when we were trying to run postfix on OpenShift. IIRC, add the account that creates the pod to the RunAsAny SCC, then set to run as uid 0 somewhere in the template. Thanks for the links, they'll surely help jog my memory to get the exact steps! :)